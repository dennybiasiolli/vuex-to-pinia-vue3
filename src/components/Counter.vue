<script setup>
import { computed } from 'vue'
import { useStore } from 'vuex'

const store = useStore()
const count = computed(() => store.state.count)
const isEven = computed(() => store.getters.isEven)
const isOdd = computed(() => store.getters.isOdd)
const increment = () => store.commit('increment')
const incrementAsync = () => store.dispatch('incrementAsync')
</script>

<template>
  <div>
    count is: {{ count }}
  </div>
  <div>
    isEven is: {{ isEven }}
    |
    isOdd is: {{ isOdd }}
  </div>
  <button @click="increment">increment sync</button>
  &nbsp;
  <button @click="incrementAsync">increment async</button>
</template>
