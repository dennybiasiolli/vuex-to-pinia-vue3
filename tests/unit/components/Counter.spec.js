import { describe, test, beforeEach, expect, vi } from 'vitest'
import { createStore } from 'vuex'
import { shallowMount } from '@vue/test-utils';
import Counter from '@/components/Counter.vue'

describe('Counter.vue', () => {
  let store;
  let mutations = {
    increment: vi.fn(),
  }
  let actions = {
    incrementAsync: vi.fn(),
  }
  beforeEach(() => {
    store = createStore({
      state: () => ({
        count: 0,
      }),
      getters: {
        isEven: () => true,
      },
      mutations,
      actions,
    })
  })
  test('should respect the snapshot', () => {
    const wrapper = shallowMount(Counter, { global: { plugins: [store] } })
    expect(wrapper.element).toMatchSnapshot()
  })

  test('increment button should call store mutation', () => {
    const wrapper = shallowMount(Counter, { global: { plugins: [store] } })
    wrapper.findAll('button')[0].trigger('click')
    expect(mutations.increment).toHaveBeenCalled()
  })

  test('incrementAsync button should call store action', () => {
    const wrapper = shallowMount(Counter, { global: { plugins: [store] } })
    wrapper.findAll('button')[1].trigger('click')
    expect(actions.incrementAsync).toHaveBeenCalled()
  })
})
