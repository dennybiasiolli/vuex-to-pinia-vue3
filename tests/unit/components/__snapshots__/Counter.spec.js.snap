// Vitest Snapshot v1

exports[`Counter.vue > should respect the snapshot 1`] = `
<div
  data-v-app=""
>
  
  <div>
     count is: 0
  </div>
  <div>
     isEven is: true | isOdd is: false
  </div>
  <button>
    increment sync
  </button>
   Â  
  <button>
    increment async
  </button>
  
</div>
`;
